Sources/Moya/Plugins/AccessTokenPlugin.swift

```diff
@@ -6,40 +6,43 @@ import Result
 /// A protocol for controlling the behavior of `AccessTokenPlugin`.
 public protocol AccessTokenAuthorizable {
 
-    /// Declares whether or not `AccessTokenPlugin` should add an authorization header
-    /// to requests.
-    var shouldAuthorize: Bool { get }
+    /// Represents the authorization header to use for requests.
+    var authorizationType: AuthorizationType { get }
+}
+
+// MARK: - AuthorizationType
+
+/// An enum representing the header to use with an `AccessTokenPlugin`
+public enum AuthorizationType: String {
+    case none
+    case basic = "Basic "
```

Comment:
The trailing space feels a bit weird. I'm assuming it's because of the formatting in:
```swift 
request.addValue(authorizationType.rawValue + tokenClosure(), forHTTPHeaderField: "Authorization")
```

If so, maybe we take the space out of the enum and properly combine `AuthorizationType` with the `tokenClosure` when adding the header?